test/quark-core/QuarkNonceManager.t.sol

```diff
@@ -353,4 +352,90 @@ contract QuarkNonceManagerTest is Test {
         );
         nonceManager.submit(nonce, true, EXHAUSTED_TOKEN);
     }
+
+    function testCancelChain() public {
+        bytes32 nonceSecret = bytes32(uint256(99));
+        bytes32 submissionToken2 = keccak256(abi.encodePacked(nonceSecret));
+        bytes32 submissionToken1 = keccak256(abi.encodePacked(submissionToken2));
+        bytes32 nonce = keccak256(abi.encodePacked(submissionToken1));
+
+        assertEq(nonceManager.submissions(address(this), nonce), FREE_TOKEN);
+
+        nonceManager.submit(nonce, true, nonce);
+        assertEq(nonceManager.submissions(address(this), nonce), nonce);
+
+        nonceManager.cancel(nonce);
```

Comment:
Would be good to test for the new `event NonceCanceled(address wallet, bytes32 nonce);` event, since it doesn't look like we're testing it anywhere yet.

Reply:
thanks; tested here: https://github.com/compound-finance/quark/pull/213/files#diff-92ae4e5be820d941323d1b2d9688aca41b944178f5726b70fa157ded4e15b693R660